service/src/main/kotlin/fi/espoo/evaka/attachment/AttachmentQueries.kt

```diff
@@ -42,15 +37,15 @@ fun Database.Transaction.insertAttachment(
         .bind("contentType", contentType)
         .let {
             when (attachTo) {
-                is AttachToApplication ->
+                is AttachmentParent.Application ->
                     it
                         .bind("applicationId", attachTo.applicationId)
                         .bindNullable("incomeStatementId", null as IncomeStatementId?)
-                is AttachToIncomeStatement ->
+                is AttachmentParent.IncomeStatement ->
                     it
                         .bindNullable("applicationId", null as ApplicationId?)
                         .bind("incomeStatementId", attachTo.incomeStatementId)
-                is AttachToNothing ->
+                is AttachmentParent.None ->
```

Comment:
Instead of binding nullables and bloating each branch when a new parent type is added, we could use `bindKotlin` and use a data class with null defaults.

```kt

data class AttachmentParent(
    val applicationId: ApplicationId? = null,
    val incomeStatementId: IncomeStatementId? = null,
    val fooId: FooId? = null,
)
```

```kt
        .bindKotlin(when (attachTo) {
            is AttachToApplication ->
                AttachmentParent(applicationId = attachTo.applicationId)
            is AttachToIncomeStatement ->
                AttachmentParent(incomeStatementId = attachTo.incomeStatementId)
            is AttachToNothing ->
                AttachmentParent()
        })
```

Reply:
Naming needs some work but you get the idea. 😄

Reply:
Great idea, fixed!